java-checks-test-sources/default/src/main/java/checks/UseMotionSensorWithoutGyroscopeCheckSample.java

```diff
@@ -0,0 +1,23 @@
+package checks;
+
+import android.hardware.Sensor;
+import android.hardware.SensorManager;
+
+public class UseMotionSensorWithoutGyroscopeCheckSample {
+
+  void compliant(SensorManager sensorManager) {
+    sensorManager.getDefaultSensor(android.hardware.Sensor.TYPE_ROTATION_VECTOR); // Noncompliant {{Replace `TYPE_ROTATION_VECTOR` (11) with `TYPE_GEOMAGNETIC_ROTATION_VECTOR` (20) to optimize battery life.}}
+    sensorManager.getDefaultSensor(11); // Noncompliant [[sc=5;ec=39]]
```

Comment:
Out of curiosity - why do you test the `[[sc=X;ec=X]]` and `{{Message}}` with two separate test cases?

Reply:
No specific reason, it may feel more readable since the comment tends to be long :)